### DISCUSSION TITLE
Run regular searches instead of being forced to prefix searches

### DISCUSSION THREAD

**Discussion by trailerparkdev** [meilisearch/product#472] [Feedback & Feature Proposal]
Hi all.

As discussed in https://github.com/meilisearch/meilisearch/issues/2378, the possibility of running a regular search instead of a prefix search would be very useful for non search-as-you-type use cases, thereby opening MeiliSearch to a much broader audience. 

You all have a nice day.

**Comment by gmourier** [comment]
Hi @trailerparkdev 👋

Reading the https://github.com/meilisearch/meilisearch/issues/2378 discussion, I'm not sure to understand what you mean by regular search.

Do you mean that every document returned contains all specified query terms instead of trying to remove the topmost right query term for each iteration to match and return documents?

Thanks!

**Comment by trailerparkdev** [comment]
Hi @gmourier.

What I meant regarding https://github.com/meilisearch/meilisearch/issues/2378 is that prefix search will yield no result if the first query term is missing in the target documents. With a "regular" search,  considering the code and example in reference, MS would yield results even if the first query (or second, etc.) term(s) is/are missing. 

In other words, if prefix search makes perfect sense in a search-as-you-type usecase, it is not operational for all other kinds of usecases. Which -- my thought was -- limits MS' possible traction and audience. Which is bad considering your good work.

Cheers.

**Reply by gmourier** [reply]
Hi @trailerparkdev 👋

Did you see this [discussion](https://github.com/meilisearch/product/discussions/255)? It makes me think that it's related to your need.

Thanks!
